core/services/payment_methods/forms/PayPalSettingsForm.php

```diff
@@ -0,0 +1,219 @@
+                        )
+                    ),
+                    'api_password' => new EE_Text_Input(
+                        array(
+                            'html_label_text' => sprintf(
+                                // translators: %s link to help doc
+                                esc_html__('API Password %s', 'event_espresso'),
+                                $help_tab_link
+                            ),
+                            'required'        => true,
+                        )
+                    ),
+                    'api_signature' => new EE_Text_Input(
+                        array(
+                            'html_label_text' => sprintf(
+                                // translators: %s link to help doc
+                                esc_html__('API Signature %s', 'event_espresso'),
+                                $help_tab_link
+                            ),
+                            'required'        => true,
+                        )
+                    ),
+                )
+            ),
+            $options_array
+        );
+        parent::__construct($options_array);
+    }
+
+    /**
+     * Tests the the PayPal API credentials work ok
+     * @return string of an error using the credentials, otherwise, if the credentials work, returns a blank string
+     * @throws EE_Error
+     */
+    protected function checkForCredentialsErrors()
+    {
+        $request_params = array(
+            'METHOD'    => 'GetBalance',
+            'VERSION'   => '204.0',
+            'USER'      => urlencode($this->get_input_value('api_username')),
+            'PWD'       => urlencode($this->get_input_value('api_password')),
+            'SIGNATURE' => urlencode($this->get_input_value('api_signature')),
+        );
+        $gateway_url = $this->get_input_value('PMD_debug_mode')
+            ? 'https://api-3t.sandbox.paypal.com/nvp'
+            : 'https://api-3t.paypal.com/nvp';
+        // Request Customer Details.
+        $response = wp_remote_post(
+            $gateway_url,
+            array(
+                'method'      => 'POST',
+                'timeout'     => 45,
+                'httpversion' => '1.1',
+                'cookies'     => array(),
+                'headers'     => array(),
+                'body'        => http_build_query($request_params, '', '&'),
+            )
+        );
+        if (is_wp_error($response) || empty($response['body'])) {
+            // If we got here then there was an error in this request.
+            // maybe is turned off. We don't know the credentials are invalid
+            EE_Error::add_error(
+                sprintf(
+                    // translators: %1$s Error message received from PayPal
+                    esc_html__(
+                        // @codingStandardsIgnoreStart
+                        'Your PayPal credentials could not be verified. There was an error communicating with PayPal, it was %1$s',
```

Comment:
Depending on how the error is worded, this could end up being really confusing, so instead of:

> There was an error communicating with PayPal, it was %1$s

how about:

> The following error occurred while communicating with PayPal: %1$s

which does not have the possibility of resulting in weird looking grammar and/or sentence structure